test/rdcycle/rdcycle.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * rdcycle.c
+ *
+ * RISC-V ISA: RV32I
+ *
+ * Copyright (C) 2017-2024 Tactical Computing Laboratories, LLC
+ * All Rights Reserved
+ * contact@tactcomplabs.com
+ *
+ * See LICENSE in the top level directory for licensing details
+ *
+ */
+
+#include "syscalls.h"
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+#define NOP1    asm( " nop" );
+#define NOP2    NOP1 NOP1
+#define NOP4    NOP2 NOP2
+#define NOP8    NOP4 NOP4
+#define NOP16   NOP8 NOP8
+#define NOP32   NOP16 NOP16
+#define NOP64   NOP32 NOP32
+#define NOP128  NOP64 NOP64
+#define NOP256  NOP128 NOP128
+#define NOP512  NOP256 NOP256
+#define NOP1024 NOP512 NOP512
+
+int main( int argc, char** argv ) {
+  size_t cycle1, cycle2;
+
+  asm volatile( " rdcycle %0" : "=r"( cycle1 ) );
+
+  // 1024 nops produces around 1024 cycles
+  NOP1024;
```

Comment:
`1024` `nop` instructions are placed between two performance counter samplings, and the difference should be between `1024` and `1026`. For `rdcycle` and `rdtime`, MemH is too slow and unpredictable. But for `rdinstret`, MemH is just as safe as RevMem`.